tests/test_coreg/test_affine.py

```diff
@@ -209,25 +319,19 @@ def test_coreg_example_shift(self, shift_px, coreg_class, points_or_raster, verb
             # The ICP fit only creates a matrix and doesn't normally show the alignment in pixels
             # Since the test is formed to validate pixel shifts, these calls extract the approximate pixel shift
             # from the matrix (it's not perfect since rotation/scale can change it).
-            coreg_obj.meta["shift_x"] = -matrix[0][3]
-            coreg_obj.meta["shift_y"] = -matrix[1][3]
+            coreg_obj.meta["outputs"]["affine"]["shift_x"] = -matrix[0][3]
+            coreg_obj.meta["outputs"]["affine"]["shift_y"] = -matrix[1][3]
 
         # ICP can never be expected to be much better than 1px on structured data, as its implementation often finds a
         # minimum between two grid points. This is clearly warned for in the documentation.
         precision = 1e-2 if coreg_class.__name__ != "ICP" else 1
 
-        if coreg_obj.meta["shift_x"] == pytest.approx(-shift_px[0] * res, rel=precision) and coreg_obj.meta[
-            "shift_y"
-        ] == pytest.approx(-shift_px[0] * res, rel=precision):
-            return
-        best_east_diff = coreg_obj.meta["shift_x"] - shift_px[0]
-        best_north_diff = coreg_obj.meta["shift_y"] - shift_px[1]
-
-        raise AssertionError(f"Diffs are too big. east: {best_east_diff:.2f} px, north: {best_north_diff:.2f} px")
+        assert coreg_obj.meta["outputs"]["affine"]["shift_x"] == pytest.approx(-shift_px[0] * res, rel=precision)
+        assert coreg_obj.meta["outputs"]["affine"]["shift_y"] == pytest.approx(-shift_px[0] * res, rel=precision)
 
     def test_nuth_kaab(self) -> None:
 
-        nuth_kaab = coreg.NuthKaab(max_iterations=10)
+        nuth_kaab = coreg.NuthKaab(max_iterations=50)
```

Comment:
is 50 really needed? Usually the algorithm converges in 2-3 iterations

Reply:
Good catch, that was me playing around while debugging :sweat_smile:

Reply:
It does converge in 2/3 iterations still